src/Auth0.OidcClient.Shared/Auth0Client.cs

```diff
@@ -194,11 +204,23 @@ public Task<LoginResult> ProcessResponseAsync(string data, AuthorizeState state)
         /// Generates a new set of tokens based on a refresh token. 
         /// </summary>
         /// <param name="refreshToken">The refresh token which was issued during the authorization flow, or subsequent
-        /// calls to <see cref="RefreshTokenAsync"/>.</param>
+        /// calls to <see cref="OidcClient.RefreshTokenAsync"/>.</param>
         /// <returns></returns>
         public Task<RefreshTokenResult> RefreshTokenAsync(string refreshToken)
         {
-            return _oidcClient.RefreshTokenAsync(refreshToken);
+            return RefreshTokenAsync(refreshToken, null);
+        }
+
+        /// <summary>
+        /// Generates a new set of tokens based on a refresh token. 
+        /// </summary>
+        /// <param name="refreshToken">The refresh token which was issued during the authorization flow, or subsequent
+        /// calls to <see cref="OidcClient.RefreshTokenAsync"/>.</param>
+        /// <param name="extraParameters">Additional parameters to send to the refresh endpoint</param>
+        /// <returns></returns>
+        public Task<RefreshTokenResult> RefreshTokenAsync(string refreshToken, object extraParameters)

```

Comment:
It looks like RefreshTokenAsync has got into this PR by mistake?  I thought it was just supposed to be the two options for the handlers.

Reply:
No I meant to include this as well. For refresh we needed to provide some additional parameters for some auth providers.(Microsoft as I remember)

Reply:
Adding a method to an interface is technically a breaking change.  I'm tempted to remove the interface entirely - it doesn't serve any useful purpose.

Reply:
You are right it is a breaking change. Hopefully only for mocks or unit tests though. It is nice to have that abstraction though. Maybe remove the interface change and just have the implementation have that change?